kernel/aster-nix/src/vm/vmar/vm_mapping.rs

```diff
@@ -216,8 +223,52 @@ impl VmMapping {
         not_present: bool,
         write: bool,
     ) -> Result<()> {
+        const AROUND_PAGE_NUM: usize = 16;
+        const AROUND_PAGE_MASK: usize = !(AROUND_PAGE_NUM * PAGE_SIZE - 1);
+
         let required_perm = if write { VmPerms::WRITE } else { VmPerms::READ };
         self.check_perms(&required_perm)?;
+
+        if !write && self.vmo.is_some() {
+            let inner = self.inner.lock();
+            if inner.need_do_pagefault_around {
```

Comment:
Inserting the do-pagefault-around logic into this methods makes the function body too large. For readability, you want to encapsulate all do-pagefault-around code into a separate method or utility or structure.